pkg/minikube/bootstrapper/bsutil/kubeadm.go

```diff
@@ -63,6 +66,9 @@ func GenerateKubeadmYAML(cc config.ClusterConfig, n config.Node, r cruntime.Mana
 
 	cgroupDriver, err := r.CGroupDriver()
 	if err != nil {
+		if driver.BareMetal(cc.Driver) && r.Name() == "Docker" && strings.Contains(err.Error(), "panic") {
+			return nil, oci.ErrDockerNotRunning
```

Comment:
this could be containerd or crio (and not docker)
so the better wording would be ErrDockerRuntimeNotRunning

and other than panic, is there a better word we could search for ?

and also lets do same for Conrtainerd and Crio
(look the implementation of  r.CGroupDriver() and manually turn of the containerd or crio (using `sudo systemctl stop containerd`  ,,,) and see what error they get ?

Reply:
Based on my experiment and the error log
```
goroutine 1 [running]:
text/template.errRecover(0xc00056b2a8)
	/usr/local/go/src/text/template/exec.go:170 +0x1b5
panic(0x5608dbf80b40, 0x5608dc2c2020)
	/usr/local/go/src/runtime/panic.go:679 +0x1b6
reflect.Value.FieldByIndex(0x5608dc139e40, 0xc0004087c0, 0x99, 0xc000500060, 0x2, 0x2, 0x0, 0x5608dc379160, 0x5608dbf80b40)
	/usr/local/go/src/reflect/value.go:883 +0x39e
text/template.(*state).evalField(0xc00056b228, 0x5608dc139e40, 0xc0004087c0, 0x99, 0x7ffd20c09c09, 0xc, 0x5608dc34a020, 0xc0002cc1b0, 0xc000298500, 0x1, ...)
	/usr/local/go/src/text/template/exec.go:605 +0x475
text/template.(*state).evalFieldChain(0xc00056b228, 0x5608dc139e40, 0xc0004087c0, 0x99, 0x5608dc139e40, 0xc0004087c0, 0x99, 0x5608dc34a020, 0xc0002cc1b0, 0xc0002984f0, ...)
	/usr/local/go/src/text/template/exec.go:558 +0x222
text/template.(*state).evalFieldNode(0xc00056b228, 0x5608dc139e40, 0xc0004087c0, 0x99, 0xc0002cc1b0, 0xc000298500, 0x1, 0x1, 0x5608dc025ac0, 0x5608dd37e5c8, ...)
	/usr/local/go/src/text/template/exec.go:522 +0x115
text/template.(*state).evalCommand(0xc00056b228, 0x5608dc139e40, 0xc0004087c0, 0x99, 0xc0002cc180, 0x5608dc025ac0, 0x5608dd37e5c8, 0x99, 0xd0, 0x5608dc119a20, ...)
	/usr/local/go/src/text/template/exec.go:460 +0x79c
text/template.(*state).evalPipeline(0xc00056b228, 0x5608dc139e40, 0xc0004087c0, 0x99, 0xc0003880c0, 0xc00057be38, 0xc0003fc0d0, 0xc00056b0e0)
	/usr/local/go/src/text/template/exec.go:434 +0x11c
text/template.(*state).walk(0xc00056b228, 0x5608dc139e40, 0xc0004087c0, 0x99, 0x5608dc349ea0, 0xc0002cc1e0)
	/usr/local/go/src/text/template/exec.go:258 +0x49d
text/template.(*state).walk(0xc00056b228, 0x5608dc139e40, 0xc0004087c0, 0x99, 0x5608dc34a0e0, 0xc0002cc150)
	/usr/local/go/src/text/template/exec.go:266 +0x144
text/template.(*Template).execute(0xc000408040, 0x5608dc2e3960, 0xc0001f5b60, 0x5608dc139e40, 0xc0004087c0, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:221 +0x20c
text/template.(*Template).Execute(...)
	/usr/local/go/src/text/template/exec.go:204
github.com/docker/cli/cli/command/system.formatInfo(0x5608dc373e80, 0xc0000c2ea0, 0x0, 0xc00033e460, 0x1, 0x1, 0xc0000d28c0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/cli/cli/command/system/info.go:490 +0x1f5
github.com/docker/cli/cli/command/system.runInfo(0xc000574840, 0x5608dc373e80, 0xc0000c2ea0, 0xc000376bf0, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/system/info.go:87 +0x37c
github.com/docker/cli/cli/command/system.NewInfoCommand.func1(0xc000574840, 0xc000217d60, 0x0, 0x2, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/system/info.go:53 +0x48
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc000574840, 0xc0005797b0, 0x2, 0x2, 0xc000574840, 0xc0005797b0)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:850 +0x462
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00022c2c0, 0xc0005797a0, 0x3, 0x3)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:958 +0x34b
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:895
main.runDocker(0xc0000c2ea0, 0x5608dc2e5f80, 0xc000010020)
	/go/src/github.com/docker/cli/cmd/docker/docker.go:287 +0x1d3
main.main()
	/go/src/github.com/docker/cli/cmd/docker/docker.go:298 +0xf3

```
**"github.com/docker/cli/cli/command/system.formatInfo"** would be more representative for docker.

Reply:
And when I tried the containerd, if containerd is running, using crictl info will get 
```
{
  "status": {
    "conditions": [
      {
        "type": "RuntimeReady",
        "status": true,
        "reason": "",
        "message": ""
      },
      {
        "type": "NetworkReady",
        "status": true,
        "reason": "",
        "message": ""
      }
    ]
  }
}
```
And if containerd is turn off, I got 
```
{
  "status": {
    "conditions": [
      {
        "type": "RuntimeReady",
        "status": false,
        "reason": "DockerDaemonNotReady",
        "message": "docker: failed to get docker version: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"
      },
      {
        "type": "NetworkReady",
        "status": true,
        "reason": "",
        "message": ""
      }
    ]
  }
}
```
So seems I have to parse the json result...